Sources/FrontEnd/Parse/Parser.swift

```diff
@@ -1714,6 +1711,29 @@ public enum Parser {
     }
   }
 
+  private static func parseFloatLiteralExpr(in state: inout ParserState) throws -> FloatLiteralExpr
+    .ID?
+  {
+    guard let first = state.take(.int) else { return nil }
+
+    if let _ = state.take(if: { $0.kind == .dot }) {
```

Comment:
```suggestion
    if state.take(if: { $0.kind == .dot }) != nil {
```
This is easier to read.